src/MainView.vala

```diff
@@ -62,30 +65,47 @@ public class DateTime.MainView : Gtk.Grid {
 
         var time_zone_label = new Gtk.Label (_("Time Zone:"));
         time_zone_label.xalign = 1;
+        time_zone_label.valign = Gtk.Align.START;
 
         auto_time_zone_icon = new Gtk.Image ();
         auto_time_zone_icon.gicon = new ThemedIcon ("location-inactive-symbolic");
+        auto_time_zone_icon.margin_start = 6;
         auto_time_zone_icon.pixel_size = 16;
 
-        var auto_time_zone_button = new Gtk.ToggleButton ();
-        auto_time_zone_button.image = auto_time_zone_icon;
-        auto_time_zone_button.tooltip_text = _("Automatically update time zone");
+        var auto_time_zone_icon_context = auto_time_zone_icon.get_style_context ();
+        auto_time_zone_icon_context.add_class ("auto-timezone-label");
+        auto_time_zone_icon_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
 
-        var css_provider = new Gtk.CssProvider ();
-        css_provider.load_from_resource ("/io/elementary/switchboard/plug/datetime/Plug.css");
+        var auto_time_zone_switch_label = new Gtk.Label (_("Based on your Location:"));
+        auto_time_zone_switch_label.margin_end = 6;
+
+        var auto_time_zone_switch = new Gtk.Switch ();
+        auto_time_zone_switch.margin_top = auto_time_zone_switch.margin_bottom = 6;
+        auto_time_zone_switch.tooltip_text = _("Automatically updates the time zone when activated");
+
+        var auto_time_zone_switch_context = auto_time_zone_switch.get_style_context ();
+        auto_time_zone_switch_context.add_class ("auto-timezone");
+        auto_time_zone_switch_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+
+        var auto_time_zone_grid = new Gtk.Grid ();
+        auto_time_zone_grid.column_spacing = 6;
+        auto_time_zone_grid.hexpand = true;
```

Comment:
Set `column_spacing` to 12 here instead of adding margins to the child widgets. Also add a bottom margin to better group this with the time zone selection above. And while we're here, update to modern code style.

```suggestion
        var auto_time_zone_grid = new Gtk.Grid () {
            column_spacing = 12,
            hexpand = true,
            margin_bottom = 12
        };

```